Sources/NIO/ByteBuffer-views.swift

```diff
@@ -80,14 +80,11 @@ extension ByteBuffer {
     ///   - index: The index the view should start at
     ///   - length: The length of the view (in bytes)
     /// - returns A view into a portion of a `ByteBuffer` or `nil` if the requested bytes were not readable.
-    public func viewBytes(at index0: Int, length: Int) -> ByteBufferView? {
-        precondition(index0 >= 0, "index must not be negative")
-        precondition(length >= 0, "length must not be negative")
-        let index = index0 - self.readerIndex
-        guard index >= 0 && index <= self.readableBytes - length else {
+    public func viewBytes(at index: Int, length: Int) -> ByteBufferView? {
+        guard index >= self.readerIndex && index <= self.writerIndex - length else {
             return nil
         }
 
-        return ByteBufferView(buffer: self, range: index0 ..< index0+length)
+        return ByteBufferView(buffer: self, range: index ..< (index + length))
```

Comment:
Can we not use the range-generation function here?

Reply:
@Lukasa `ByteBufferView` uses (and has to use as `Collection`) absolute ByteBuffer indices so here, we're not interested in range relative to readable bytes but range relative to the whole ByteBuffer

Reply:
Ah, yeah, I see that.

This represents a difficulty with the requirement that `ByteBufferView`s only index readable bytes: specifically, what is readable can *change* after the `ByteBufferView` is created, without invalidating the `ByteBufferView` itself. I don't think this particularly matters much, as this will only happen if you're going outside the standard NIO usage patterns, but it's worth being aware of.

Reply:
@Lukasa no, `ByteBufferView` holds a (private) `ByteBuffer` which is a value, so the readable bytes can't change.